collections/_article/progress-report-web-export-in-4-3.md

```diff
@@ -0,0 +1,311 @@
+Golden Age of Web games. Not only websites like [Poki](https://poki.com/) or
+[Crazy Games](https://www.crazygames.com/) are super popular, but big players are starting to integrate Web games into
+their services, such as
+[Discord Activities](https://support.discord.com/hc/en-us/articles/4422142836759-Activities-on-Discord)
+or [YouTube Playables](https://www.youtube.com/playables). And all these entities want developers to create games for
+their platform. And they are all asking how Godot could help create these games.
+
+That was an eye opening moment.
+
+I knew that making games for the Web using Godot was a challenge, especially on 4.x, and that serious work was needed.
+You can ask game jam participants that want their games playable on [itch.io](https://itch.io/), as this
+greatly help people play your game.
+
+I always considered Web games to be an extension of existing desktop games, though.
+You made a game, and you can port it to the Web if you want. Worked fine (with said caveats) so far.
+
+But "instant games" are more popular than ever, and their growth is just beginning. Web games promise quality gaming
+without having to download anything. No apps, no purchases needed.
+
+Fortunately, I already had some work done towards fixing some of the long lasting issues with Godot 4.x regarding Web
+exports. The <abbr title="Game Developers Conference">GDC</abbr> just hammered very hard on the fact that the Web
+platform to-do list was way longer than I imagined.
+
+# Godot 4.3
+Godot 4.3 promises to be one of the best recent releases about Web exports. One of the biggest issues relating to this
+has been properly fixed: single-threaded exports.
+
+## Single-threaded Web export
+
+### Betting on the wrong horse
+[`SharedArrayBuffer`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer)s
+were supposed to revolutionize the Web. And they did. That API permits to share memory between
+[Web Workers](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Using_web_workers)
+(which are the Web native "threads").
+
+Unfortunately, we live in the same timeline that of [Spectre and Meltdown exploits](https://meltdownattack.com/).
+
+The result of this is that browsers greatly reduced where and how you can use that API. Namely, browsers nowadays require
+websites to be [cross-origin isolated](https://web.dev/coop-coep/). When isolated, it does unlock the potential of
+`SharedArrayBuffer`s, but at what cost? The capacity to make remote calls to other websites.
+
+_Adieu_ game monetization. _Adieu_ interacting with an external API.
+
+!["What did it cost? Everything." exchange between Gamora and Thanos in Avengers Infinity](/storage/blog/progress-report-web-export-in-4-3/thanos-gamora.gif)
+
+The thing is that during the development of Godot 4, misconceptions about the Web platform made it so that we bet on
```

Comment:
What misconceptions? As someone not aware this only opens up questions instead of answering them

Reply:
Is this better?
> The thing is that during the development of Godot 4, we bet on the wrong horse: we went all-in with using threads with `SharedArrayBuffer`s.

Reply:
Yes, but maybe "The issue is" to make it a tad more formal ;)